File: java/growList.java
import java.util.LinkedList;

/**
 * Algorithm which pushes integers to a dynamic (growing) array (here LinkedList)
 */
class growList {
  public static void main(String[] args) {
    long startTime;
    int maxNum = 1000000; // by default 1 million integers in array
    LinkedList<Integer> intArray = new LinkedList<Integer>();

    if (args.length != 0) {
      System.out.println("Parameter (String) provided, parsing to integer: " + args[0]);
      maxNum = Integer.parseInt( args[0] );
    } else {
      System.out.println("No parameter provided, using default " + maxNum);
    }

    startTime = System.nanoTime();

    for (int i = 0; i < maxNum; i++) {
      intArray.addLast(i);
    }

    System.out.println(
      "Time " + (float) (System.nanoTime() - startTime) / 1000000000 + "s"
    );
  }
}

/*
Intel i7 2.9 GHz, Windows 10 64bit, Java OpenJDK 15.0.1

3 test runs with parameters: 1000002, 1000003, 1000004 (to avoid any runtime caching)

0.1172112
0.1074784
0.1174478
= AVERAGE: 114.0ms

*/

File: cs/growList.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;

/**
 * Algorithm which pushes integers to a dynamic (growing) array, here using generic List<T>
 */
class GrowList
{
  static void Main(string[] args)
  {
    List<int> intArray = new List<int>();
    int maxNum = 1000000; // by default 1 million integers in array
    Stopwatch stopwatch = new Stopwatch();

    if (args.Length != 0)
    {
      Console.WriteLine("Parameter (string) provided, parsing to integer: " + args[0]);
      maxNum = Int32.Parse(args[0]);
    }
    else
    {
      Console.WriteLine("No parameter provided, using default {0}", maxNum);
    }

    stopwatch.Start();

    for (int i = 0; i < maxNum; i++)
    {
      intArray.Add(i);
    }

    stopwatch.Stop();
    Console.WriteLine("Time elapsed: {0}{1}", stopwatch.ElapsedMilliseconds, "ms");
  }
}

/*
Intel i7 2.9 GHz, Windows 10 64bit
3 test runs with parameters: 1000002, 1000003, 1000004

C# v8.0 (DotNet Core)
Time elapsed: 6ms
Time elapsed: 7ms
Time elapsed: 6ms
AVERAGE: 6.3ms

Speed with C# 9 (.NET 5) is equivalent.
*/
